app/implementation_handler.py

```diff
@@ -85,8 +86,19 @@ def _download_code(url: str, bearer_token: str = "") -> str:
     req = request.Request(url)
 
     if urllib.parse.urlparse(url).netloc == "platform.planqk.de":
-        req.add_header("Authorization", bearer_token)
+        if bearer_token == "":
+            app.logger.error("No bearer token specified, download from the PlanQK platform will fail.")
+        elif bearer_token.startswith("Bearer"):
+            app.logger.error("The bearer token MUST NOT start with \"Bearer\".")
+
+        req.add_header("Authorization", "Bearer " + bearer_token)
+
+    try:
+        res: HTTPResponse = request.urlopen(req)
+    except Exception as e:
+        app.logger.error("Could not open url: " + str(e))
```

Comment:
do not return a default 500 internal server error in case of catching the error. I think 401 Unauthorized suits best.